coordinator/src/main/scala/filodb/coordinator/v2/NewNodeCoordinatorActor.scala

```diff
@@ -30,6 +29,36 @@ object NewNodeCoordinatorActor {
             clusterDiscovery: FiloDbClusterDiscovery,
             settings: FilodbSettings): Props =
     Props(new NewNodeCoordinatorActor(memStore, clusterDiscovery, settings))
+
+
+  /**
+   * Converts a ShardMapper.statuses to a bitmap representation where the bit is set to:
+   *  - 1, if ShardStatus == ShardStatusActive
+   *  - 0, any other ShardStatus like ShardStatusAssigned, ShardStatusRecovery etc.
+   *  WHY this is the case ? This is because, the QueryActor is can only execute the query on the active shards.
+   *
+   * NOTE: bitmap is byte aligned. So extra bits are padded with 0.
+   *
+   * EXAMPLE - Following are some example of shards with statuses and their bit representation as below:
+   *                   Status                          |  BitMap Representation      |  Hex Representation
+   *  ---------------------------------------------------------------------------------------------------------------
+   *  Assigned, Active, Recovery, Error                |      0100 0000              |      0x40
```

Comment:
so, most significant bit is shard 0 (in case if number of shards is divisible by 8)?

Reply:
Yes, in any case shard 0 is the most significant bit , just like array notation.